packages/contracts-bedrock/scripts/deploy/Deploy.s.sol

```diff
@@ -407,21 +407,47 @@ contract Deploy is Deployer {
         deployMulticall3();
     }
 
+    function preInitializeOptimismPortalBalance() public broadcast {
+        address optimismPortalProxy = mustGetAddress("OptimismPortalProxy");
+        address storageSetter = mustGetAddress("StorageSetter");
+
+        // NOTE: the storage slot index should stay the same across versions
+        // (OptimismPortal, OptimismPortal2, ...)  since slot spacers are used
+        // for legacy storage variables.
+        // We also assert correctness in a downstream ChainAssertion,
+        // so changing slot numbers should get detected for coming versions.
+        uint256 balanceStorageSlot = 61; // slot of _balance variable
+
+        address customGasTokenAddress = Constants.ETHER;
+        uint256 initialBalance = 0;
+        customGasTokenAddress = cfg.customGasTokenAddress();
+        IERC20 token = IERC20(customGasTokenAddress);
+        initialBalance = token.balanceOf(optimismPortalProxy);
```

Comment:
Where is the balance of the optimismPortalProxy set?

Reply:
This is set within the `Deploy.s.sol` script:

Our Celo L1 Token contract takes the Portal address as an initializer argument:
https://github.com/celo-org/optimism/blob/981f87fa97e5c65c8b5ebe4fd589fa6320a160f3/packages/contracts-bedrock/scripts/deploy/Deploy.s.sol#L1693-L1697

And then sets the market cap as it's value:
https://github.com/celo-org/optimism/blob/981f87fa97e5c65c8b5ebe4fd589fa6320a160f3/packages/contracts-bedrock/src/celo/CeloTokenL1.sol#L8-L14